### PR TITLE
Refactor Scout/parse Functions Reducing Complexity (#2984)

### PR DESCRIPTION
**Pull request by mikaell**
Refactor Parse/ensembl.py according to #2984.

**How to test**:
1. how to test it, possibly with real cases/data

Refactor PR. Functionality should not have changed. 

Pytests should pass. 

Test complexity with:
```
radon cc scout/parse/ensembl.py scout/parse/variant/frequency.py scout/parse/variant/genotype.py

```
Master will be more complex.
image

**Expected outcome**:
The functionality should be working. Pytests should pass. 

Function complexity is reduced. See screenshot.

image

**Review:**
- [x] code approved by DN
- [x] tests executed by GitHub Actions

### CODE REVIEW

**Comment by mikaell** [comment]
When tests pass this is ready for a review.

File: CHANGELOG.md
Comment (northwestwitch) [inline CHANGELOG.md:16]: Nice! Don't forget to check the relative checkboxes in #2984, so we know which functions have been rewritten!

Comment (mikaell) [reply CHANGELOG.md:16]: Will do!

File: scout/parse/variant/transcript.py
Comment (dnil) [inline scout/parse/variant/transcript.py:24]: Agreed, looks really useless? The data is stored in functional_annotations, and there should be no way this function terminates in an orderly fashion before the point where the transcript reference to it is overwritten! Let's remove this assignment.

Vulture also notes a few unused above - seems safe to fix.

File: scout/parse/variant/transcript.py
Comment (dnil) [inline scout/parse/variant/transcript.py:32]: 👍

File: scout/parse/variant/transcript.py
Comment (dnil) [inline scout/parse/variant/transcript.py:42]: Just for uniformity with the SIFT below.
```suggestion
        transcript["polyphen_prediction"] = get_polyphen_prediction(entry)
```

File: scout/parse/variant/transcript.py
Comment (dnil) [inline scout/parse/variant/transcript.py:244]:
```suggestion
def get_polyphen_prediction(entry):
```

File: scout/parse/variant/transcript.py
Comment (dnil) [inline scout/parse/variant/transcript.py:253]: docstring!

File: scout/parse/variant/transcript.py
Comment (dnil) [inline scout/parse/variant/transcript.py:120]: update docstring or function name instead of adding comment!

File: scout/parse/variant/transcript.py
Comment (dnil) [inline scout/parse/variant/transcript.py:132]: ✅

File: scout/parse/variant/transcript.py
Comment (dnil) [inline scout/parse/variant/transcript.py:133]: ✅

File: scout/parse/variant/transcript.py
Comment (dnil) [inline scout/parse/variant/transcript.py:75]: ✅

File: scout/parse/variant/transcript.py
Comment (dnil) [inline scout/parse/variant/transcript.py:213]: Redundant else

File: scout/parse/variant/transcript.py
Comment (dnil) [inline scout/parse/variant/transcript.py:79]: Choose if you want to pass transcript, and update it, or take the return value! Its not currently used in the function.

**Review by dnil — CHANGES_REQUESTED** [review]
Super! Thank you kindly @mikaell for working on this. I found some small things here and there, but this will make it much easier to find stuff in the future! I'll hit request changes just because its so many places so its good to get another pair of eyes on it before merging. When you do more of these, consider several smaller PRs!

File: scout/parse/ensembl.py
Comment (mikaell) [reply scout/parse/ensembl.py:42]: You are not missing something. I think it's my habits shining through, that a function will return a value 🙂
Maybe it should be rewritten to a more python-like:

```
update_something(x,y)
update_somethingelse(x,z)
...
```

File: scout/parse/variant/transcript.py
Comment (mikaell) [reply scout/parse/variant/transcript.py:213]: Thanks, will remove `else` but keep `return None` (I know that the return would be redundant as well, but somewhere you've got to draw the line).

File: scout/parse/variant/transcript.py
Comment (mikaell) [reply scout/parse/variant/transcript.py:79]: Yes. Will do.

File: scout/parse/ensembl.py
Comment (dnil) [reply scout/parse/ensembl.py:42]: Right, it is mind boggling to a Python non-native with a function named "read_" or "get_" that modifies its arguments and has no caught return value. Having lived a long time with C/C++, I mentally imagine pointer characters on appropriate variables in python, but its still weird. I have no objections to changing the names, but it does not seem mandatory either!

Comment (mikaell) [reply scout/parse/ensembl.py:42]: Added a function checking for substrings and which returns true if all are present. Turned the code to something like this:

```
        if has_substrings(["utr", "start", "5"], word):
            ensembl_info["utr_5_start"] = int(value)
        elif has_substrings(["utr", "start", "3"], word):
            ensembl_info["utr_3_start"] = int(value)
        elif has_substrings(["utr", "end", "5"], word):
            ensembl_info["utr_5_end"] = int(value)
        elif has_substrings(["utr", "end", "3"], word):
            ensembl_info["utr_3_end"] = int(value)

```

File: scout/parse/ensembl.py
Comment (mikaell) [reply scout/parse/ensembl.py:33]: I choose to extract operations two construct levels or deeper, here is only one if-clause. Ok to leave it?

File: scout/parse/ensembl.py
Comment (mikaell) [reply scout/parse/ensembl.py:42]: Above became less readable. Will update the dict and never return it. Soon to be pushed for review.

File: scout/parse/ensembl.py
Comment (dnil) [reply scout/parse/ensembl.py:33]: Sure sure, It would just fit the function scope and do kind of the same thing, but its already better than before so no worries!

File: scout/parse/variant/transcript.py
Comment (mikaell) [reply scout/parse/variant/transcript.py]: Done!

File: scout/parse/variant/transcript.py
Comment (mikaell) [reply scout/parse/variant/transcript.py:79]: Done!

**Comment by mikaell** [comment]
This is ready for a new review. But Woke seems to fail on me. Don't know what I'm doing wrong.

**Comment by dnil** [comment]
> This is ready for a new review. But Woke seems to fail on me. Don't know what I'm doing wrong.

Maybe it was something temporary? It worked yesterday, and when I triggered it now (😜) it worked just fine.

Great anyway, I'll have a look!

**Comment by dnil** [comment]
> This is ready for a new review. 

There still seem to be a couple of items from last review that were not addressed? Let me know if you want some help with them!

File: scout/parse/variant/transcript.py
Comment (mikaell) [reply scout/parse/variant/transcript.py:120]: Done!

File: scout/parse/variant/transcript.py
Comment (mikaell) [reply scout/parse/variant/transcript.py:243]: Done!

File: scout/parse/variant/transcript.py
Comment (mikaell) [reply scout/parse/variant/transcript.py:252]: Done!

File: scout/parse/variant/transcript.py
Comment (mikaell) [inline scout/parse/variant/transcript.py:299]: The function `get_functional_annotations` is a waste of bits .

**Review by dnil — APPROVED** [review]
Nice, thank you @mikaell! I took the liberty to fix one more gripe from CodeFactor that seemed kind of wise.

**Pull request merged**
